pkg/placement/raft/state.go

```diff
@@ -77,53 +97,73 @@ type DaprHostMemberStateConfig struct {
 func newDaprHostMemberState(config DaprHostMemberStateConfig) *DaprHostMemberState {
 	return &DaprHostMemberState{
 		config: config,
-		data: DaprHostMemberStateData{
-			Members:         map[string]*DaprHostMember{},
-			hashingTableMap: map[string]*hashing.Consistent{},
-		},
+		data:   newDaprHostMemberStateData(),
 	}
 }
 
 func (s *DaprHostMemberState) Index() uint64 {
-	s.lock.RLock()
-	defer s.lock.RUnlock()
+	s.Lock.RLock()
+	defer s.Lock.RUnlock()
 
 	return s.data.Index
 }
 
 // APILevel returns the current API level of the cluster.
 func (s *DaprHostMemberState) APILevel() uint32 {
-	s.lock.RLock()
-	defer s.lock.RUnlock()
+	s.Lock.RLock()
+	defer s.Lock.RUnlock()
 
 	return s.data.APILevel
 }
 
-func (s *DaprHostMemberState) Members() map[string]*DaprHostMember {
-	s.lock.RLock()
-	defer s.lock.RUnlock()
+// Namespaces requires a lock to be held by the caller.
+func (s *DaprHostMemberState) Namespaces() []string {
+	namespaces := make([]string, 0, len(s.data.Namespace))
+	for ns := range s.data.Namespace {
+		namespaces = append(namespaces, ns)
+	}
 
-	members := make(map[string]*DaprHostMember, len(s.data.Members))
-	for k, v := range s.data.Members {
-		members[k] = v
+	return namespaces
+}
+
+// Members requires a lock to be held by the caller.
+func (s *DaprHostMemberState) Members(ns string) (map[string]*DaprHostMember, error) {
+	n, ok := s.data.Namespace[ns]
```

Comment:
I see :slightly_smiling_face: 

Callers needing to manage locking can be quite fragile to future refactoring- are we sure we don't want to do the locking in this func? Generally- memory locking should be solely managed by the package itself.

Reply:
I added a few methods to DaprHostMemberState, and now we don't have to export the lock 🙏